Automattic/pocket-casts-ios
# Referrals: Dismiss hints and banner on user actions

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR changes the following:

- Dismiss referral hint on tap of the tip
- Show claim referral banner immediately on referral url link tap
- Hide claim banner when tapping close

## To test

1. Start the app ( Delete any version you had before on the device)
2. Ensure that you have the `referrals` FF enabled
3. Ensure that you have enabled logging for Tracks events

### Scenario 1 - Referrals hints dismiss
1. Login with an account that has Plus or Patron
2. Open the Profile tab
4. Check that you see the referral hint on the top left 
5. Move to another tab and see if the hint/tip goes away
6. Move back to Profile tab
7. Tap on the hint
8. See if it's dismissed and you see the event on the console: `referral_tooltip_tapped`
9. Tap on the Gift icon on the top left
10. Tap on `Share Guest Pass` and save the referral URL to Reminders

### Scenario 2 - Referrals Banner dismiss

1. Login with an account that does not have Plus or Patron
2. Tap on the referral link on  Reminders
3. The app should open the claim pass screen
4. Tap on not now
5. You should see the banner on the Profile View
6. Tap on the X close button
7. Check if the banner is dismissed

## Checklist
